phac-nml/irida-next
# Pipelines Configuration: Parameter options overrides (DFCT0010053)

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Fixes #592 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

In  `config/pipelines/pipelines.json` add mikrokonda:

```json
{
    "url": "https://github.com/phac-nml/mikrokondo",
    "name": "phac-nml/mikrokondo",
    "description": "Mikrokondo pipeline",
    "parameter_overrides": {
      "definitions": {
        "databases_and_pre_computed_files": {
          "properties": {
            "kraken2_db": {
              "type": "string",
              "description": "Kraken2 database",
              "enum": [
                [
                  "DBNAME",
                  "PATH_TO_DB"
                ],
                [
                  "ANOTHER_DB",
                  "ANOTHER_PATH"
                ]
              ]
            }
          }
        }
      }
    },
    "versions": [
      {
        "name": "0.1.2"
      }
    ]
  }
```

After selecting the pipeline, it should render:

![image](https://github.com/phac-nml/irida-next/assets/11295750/b9f68b7a-1571-459e-bba3-412c43dcc8d5)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
